src/viper/trans.ml

```diff
@@ -30,8 +30,30 @@ let rec unit (u : Mo_def.Syntax.comp_unit) : prog =
   match body.it with
   | M.ActorU(id_opt, decs) ->
      let ctxt = { self = None; ids = Env.empty } in
-     let (ctxt', mk_is) = dec_fields ctxt decs in
-     let is = List.map (fun mk_i -> mk_i ctxt') mk_is in
+     let ctxt', inits, mk_is = dec_fields ctxt decs in
+     let is' = List.map (fun mk_i -> mk_i ctxt') mk_is in
+
+     let init_id = {it = "__init__"; at = Source.no_region; note = NoInfo } in
+     let self_id = {it = "$Self"; at = Source.no_region; note = NoInfo } in
+     let ctxt'' = { ctxt' with self = Some self_id.it } in
+     let init_list = (List.map (fun (id, init) -> {
+      at = init.at;
```

Comment:
Maybe join to an initialiser and have `{ left = id.at.left; right = init.at.right }`?

Reply:
Could you clarify what you mean, please?

Reply:
To me it looks like the `FieldAssignS` inherits the source range from the `init` (value). I am suggesting to give it the source range from the left of the identifier (field name) until the right of the `init` value.

Reply:
Thanks for the clarification. I didn't know we already supported `left` and `right` in source positions. This makes sense!